src/main/scala/machines/Internal.scala

```diff
@@ -3,4 +3,44 @@ package machines
 import regex._
 import dfa._
 
-// TODO: Add your code below
+// Part 1:
+given Conversion[Char, RegularLanguage] = Character(_)
+
+
+// Part 2:
+given Conversion[String, RegularLanguage] = (s: String) => {
+  val chars = s.map(Character(_))
+  chars.reduce((a, b) => Concat(a, b))
```

Comment:
Instead of using map, I used Scala's toList function to make the string into a list of chars. I think this probably makes the code a bit more readable because the naming makes its function clearer. I also used fold right instead of reduce but in this case, I think yours is more concise as mine reads like this: charList.foldRight(Concat(Epsilon, Epsilon))(Concat(_, _))

Reply:
I did the same thing. This version is a little unreadable but honestly I wouldn't change it. Also its a bit more future proof